pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java

```diff
@@ -171,4 +177,49 @@ public void testHealthCheckTopicNotOffload() throws Exception {
         });
     }
 
+    @Test
+    private void testSetBacklogCausedCreatingProducerFailure() throws Exception {
+        final String ns = "prop/ns-test";
+        final String topic = ns + "/topic-1";
+
+        admin.namespaces().createNamespace(ns, 2);
+        admin.topics().createPartitionedTopic(String.format("persistent://%s", topic), 1);
+        BacklogQuota quota = BacklogQuota.builder()
+                .limitTime(30)
+                .limitSize(-1)
+                .retentionPolicy(BacklogQuota.RetentionPolicy.producer_exception)
+                .build();
+        admin.namespaces().setBacklogQuota(ns, quota, BacklogQuota.BacklogQuotaType.message_age);
+
+        @Cleanup
+        Producer<String> producer = pulsarClient.newProducer(Schema.STRING)
+                .topic(topic)
+                .create();
+        String msg1 = "msg-1";
+        producer.send(msg1);
+        Thread.sleep(60 * 1000);
```

Comment:
Could you avoid sleeping for such a long time?

Reply:
Minimum rollover is 1m

Reply:
I think we can get the `ManagedLedger` object form the `PersistentTopic`, then change the rollover time manually.

```java
        final ManagedLedgerImpl managedLedger = pulsar.getBrokerService().getTopicIfExists(partitionName).get()
                .map(topicObject -> (ManagedLedgerImpl) ((PersistentTopic) topicObject).getManagedLedger())
                .orElse(null);
        managedLedger.getConfig()./* change rollover time... */;
```

Reply:
ok

Reply:
fixed.